File: Portfolio.Domain/Models/Dissertation.cs
using System.Text.Json.Serialization;

namespace Portfolio.Domain.Models
{
    public enum DissertationType
    {
        None,
        Master,
        Candidate,
        Doctor
    }
    public class Dissertation
    {
        public Guid Id { get; set; }
        public DissertationType Type { get; set; }
        public int YearProtection { get; set; }
        public string Topic { get; set; }
        public string Specialization { get; set; }
        public Guid TeacherId { get; set; }
        [JsonIgnore]
        public Teacher Teacher { get; set; }
    }
}

File: Portfolio.Frontend/src/classes/Dissertation.ts
import { DissertationType } from "@/enums/DissertationType";

export class Dissertation {
  id: string;
  yearProtection: number;
  type: DissertationType;
  specialization: string;
  topic: string;

  constructor(data: Partial<Dissertation> = {}) {
    this.id = data.id || '';
    this.yearProtection = data.yearProtection || new Date().getFullYear();
    this.type = data.type || DissertationType.Master;
    this.specialization = data.specialization || '';
    this.topic = data.topic || '';
  }
}
